### PR TITLE
Term Entry: Nmap TCP Connect Scan

### PR DESCRIPTION
**Pull request by joesizzlin**
### Description

Closes  [Term Entry] Nmap: TCP Connect Scan #1315 

Wrote an entry for [Nmap ](https://www.codecademy.com/resources/docs/cybersecurity/nmap)about the TCP Connect Scan option.
Located under docs/content/cybersecurity/concepts/nmap/terms/tcp-connect-scan/tcp-connect-scan.md

Includes an image which is added and linked to docs/media/network_scan_01.png

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

**Comment by joesizzlin** [comment]
Closes [Term Entry] Nmap: TCP Connect Scan https://github.com/Codecademy/docs/issues/1315

File: content/cybersecurity/concepts/nmap/terms/tcp-connect-scan/tcp-connect-scan.md
Comment (SSwiniarski) [inline content/cybersecurity/concepts/nmap/terms/tcp-connect-scan/tcp-connect-scan.md:3]:
```suggestion
Description: 'A type of Nmap scan that establishes a complete connection to the target.'
```

File: content/cybersecurity/concepts/nmap/terms/tcp-connect-scan/tcp-connect-scan.md
Comment (SSwiniarski) [inline content/cybersecurity/concepts/nmap/terms/tcp-connect-scan/tcp-connect-scan.md:16]:
```suggestion
A **TCP Connect Scan** (`-sT`) is a type of Nmap scan that establishes a complete connection to the target by completing a TCP 3-way handshake. 
```

File: content/cybersecurity/concepts/nmap/terms/tcp-connect-scan/tcp-connect-scan.md
Comment (SSwiniarski) [inline content/cybersecurity/concepts/nmap/terms/tcp-connect-scan/tcp-connect-scan.md:20]:
```suggestion
This is not to be confused with another common scan technique, TCP Syn Scan (`-sS`) which does not complete its connection to the target host. Because of this, TCP Connect Scan (`-sT`) will take longer and require more packets to perform.
```

File: content/cybersecurity/concepts/nmap/terms/tcp-connect-scan/tcp-connect-scan.md
Comment (SSwiniarski) [inline content/cybersecurity/concepts/nmap/terms/tcp-connect-scan/tcp-connect-scan.md:25]: This needs to be a full path to where the image will live once this is merged.

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@joesizzlin I did a first review and left a few comments.

File: content/cybersecurity/concepts/nmap/terms/tcp-connect-scan/tcp-connect-scan.md
Comment (SSwiniarski) [reply content/cybersecurity/concepts/nmap/terms/tcp-connect-scan/tcp-connect-scan.md:25]: @Dusch4593, @yangc95 how do we feel about reusing existing Codecademy assets?

**Review by SSwiniarski — CHANGES_REQUESTED** [review]

File: content/cybersecurity/concepts/nmap/terms/tcp-connect-scan/tcp-connect-scan.md
Comment (joesizzlin) [reply content/cybersecurity/concepts/nmap/terms/tcp-connect-scan/tcp-connect-scan.md]: @SSwiniarski I addressed this suggestion between lines 23 and 25. I believe the correct full path is now here and correct?
I am not opposed to omitting the image from this pull request and addressing in a different branch, provided you and the rest of the team are okay with reusing Codecademy assets?

Comment (yangc95) [reply content/cybersecurity/concepts/nmap/terms/tcp-connect-scan/tcp-connect-scan.md]: @SSwiniarski I think if they're reused, we should stick to the format we already have where they're stored in the media folder. I see @joesizzlin has gone ahead and followed that pattern.

I would like to see a full explanation for the linked image though. Since this is documentation, a description should be given 🙂

Comment (joesizzlin) [reply content/cybersecurity/concepts/nmap/terms/tcp-connect-scan/tcp-connect-scan.md]: @SSwiniarski, @yangc95  I am drafting up a brief description for the linked image so that way it is ready to go if we need it. 

I plan to sit on this briefly to be sure there is full agreement: 
1. the image is now linked correctly 
2. if we want to keep the image in the documentation

**Review by SSwiniarski — APPROVED** [review]
@joesizzlin looks good for a second review.

File: content/cybersecurity/concepts/nmap/terms/tcp-connect-scan/tcp-connect-scan.md
Comment (Dusch4593) [inline content/cybersecurity/concepts/nmap/terms/tcp-connect-scan/tcp-connect-scan.md:25]:
```suggestion
  ![TCP handshake and network scan](https://static-assets.codecademy.com/skillpaths/learn-nmap/Performing_Basic_Network_Scans/network_scan_01.png)
```

Since the image we're using is already a static asset file within Codecademy's content base, we actually don't need to save a copy and store in the `media` folder! We can just use the `static-assets` link directly and it should render fine. So let's remove the former link along with the `network_scan_01.png` file stored in `media`. 😄

File: content/cybersecurity/concepts/nmap/terms/tcp-connect-scan/tcp-connect-scan.md
Comment (Dusch4593) [inline content/cybersecurity/concepts/nmap/terms/tcp-connect-scan/tcp-connect-scan.md:3]:
```suggestion
Description: 'Establishes a complete connection to the target host.'
```

File: content/cybersecurity/concepts/nmap/terms/tcp-connect-scan/tcp-connect-scan.md
Comment (Dusch4593) [inline content/cybersecurity/concepts/nmap/terms/tcp-connect-scan/tcp-connect-scan.md:20]:
```suggestion
A **TCP connect scan** establishes a complete connection to the target host by completing a TCP three-way handshake. After the scan is complete, Nmap terminates the connection.

> **Note:** This is not to be confused with another common scan technique, the TCP SYN stealth scan (`-sS`), which completes up to half of its connection with the target host. Therefore, the TCP connect scan takes longer and requires more packets to perform.
```

File: content/cybersecurity/concepts/nmap/terms/tcp-connect-scan/tcp-connect-scan.md
Comment (Dusch4593) [inline content/cybersecurity/concepts/nmap/terms/tcp-connect-scan/tcp-connect-scan.md:24]:
```suggestion
The following image shows how this scan connects with the TCP three-way handshake and then terminates afterward: 
```

File: content/cybersecurity/concepts/nmap/terms/tcp-connect-scan/tcp-connect-scan.md
Comment (Dusch4593) [inline content/cybersecurity/concepts/nmap/terms/tcp-connect-scan/tcp-connect-scan.md:33]:
```suggestion
The `-sT` flag is used to perform a TCP connect scan on  a `<target>` host (e.g., a [URL](https://www.codecademy.com/resources/docs/general/url) or IP address).
```

File: content/cybersecurity/concepts/nmap/terms/tcp-connect-scan/tcp-connect-scan.md
Comment (Dusch4593) [inline content/cybersecurity/concepts/nmap/terms/tcp-connect-scan/tcp-connect-scan.md:37]:
```suggestion
The following example performs a TCP connect scan on the site [scanme.nmap.org](http://scanme.nmap.org/):
```

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@joesizzlin Left you some comments/suggestions 😄

**Comment by joesizzlin** [comment]
@Dusch4593 thanks for reviewing! I will chip away those suggestions over the next several days :smiley:

File: content/cybersecurity/concepts/nmap/terms/tcp-connect-scan/tcp-connect-scan.md
Comment (joesizzlin) [reply content/cybersecurity/concepts/nmap/terms/tcp-connect-scan/tcp-connect-scan.md:3]: "Present-tense verb" got it! :smiley:

File: content/cybersecurity/concepts/nmap/terms/tcp-connect-scan/tcp-connect-scan.md
Comment (joesizzlin) [reply content/cybersecurity/concepts/nmap/terms/tcp-connect-scan/tcp-connect-scan.md:23]: This is great to know! Thank you :smiley:

Comment (joesizzlin) [reply content/cybersecurity/concepts/nmap/terms/tcp-connect-scan/tcp-connect-scan.md]: removed `network_scan_01.png` :boom:

File: content/cybersecurity/concepts/nmap/terms/tcp-connect-scan/tcp-connect-scan.md
Comment (joesizzlin) [reply content/cybersecurity/concepts/nmap/terms/tcp-connect-scan/tcp-connect-scan.md:20]: I like your consolidation here

**Review by joesizzlin — COMMENTED** [review]
@Dusch4593  All suggestions in `tcp-connect-scan.md` looked great and are committed.
`network_scan_01.png` removed from `media`
Thank you :smiley:

File: content/cybersecurity/concepts/nmap/terms/tcp-connect-scan/tcp-connect-scan.md
Comment (Dusch4593) [inline content/cybersecurity/concepts/nmap/terms/tcp-connect-scan/tcp-connect-scan.md:20]:
```suggestion
The following image shows how this scan connects with the TCP three-way handshake and then terminates afterward:

```

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@joesizzlin On line # 31, can we change to the following?

> A TCP connect scan connects to a `<target>` host (e.g., a [URL](https://www.codecademy.com/resources/docs/general/url) or IP address).

File: content/cybersecurity/concepts/nmap/terms/tcp-connect-scan/tcp-connect-scan.md
Comment (joesizzlin) [inline content/cybersecurity/concepts/nmap/terms/tcp-connect-scan/tcp-connect-scan.md:33]:
```suggestion
A TCP connect scan connects to a <target> host (e.g., a [URL](https://www.codecademy.com/resources/docs/general/url) or IP address).
```

File: content/cybersecurity/concepts/nmap/terms/tcp-connect-scan/tcp-connect-scan.md
Comment (joesizzlin) [inline content/cybersecurity/concepts/nmap/terms/tcp-connect-scan/tcp-connect-scan.md:29]:
```suggestion
A TCP connect scan connects to a <target> host (e.g., a [URL](https://www.codecademy.com/resources/docs/general/url) or IP address).
```

**Review by joesizzlin — COMMENTED** [review]
@Dusch4593 I think I understood everything and made the necessary corrections :smiley:

**Review by Dusch4593 — APPROVED** [review]
@joesizzlin LGTM! Made a small edit in 71f26b94b8b3aef4c4e4ee23f06b0d35c7d2a312. Looks ready to merge. 😄

**Comment by Dusch4593** [comment]
Thank you for another awesome contribution to Codecademy Docs, @joesizzlin! 🚀 🎈 🎉 🥳 

The new entry is now live: https://www.codecademy.com/resources/docs/cybersecurity/nmap/tcp-connect-scan

Screen Shot 2023-01-18 at 7 45 56 PM

Looking forward to your next contribution to Docs! 😄

**Pull request merged**
